pkg/flow/internal/controller/config.go

```diff
@@ -34,3 +37,128 @@ func NewConfigNode(block *ast.BlockStmt, globals ComponentGlobals, isInModule bo
 		return nil, diags
 	}
 }
+
+// ConfigNodeMap represents the config BlockNodes in their explicit types.
+// This is helpful when validating node conditions specific to config node
+// types.
+type ConfigNodeMap struct {
+	logging     *LoggingConfigNode
+	tracing     *TracingConfigNode
+	argumentMap map[string]*ArgumentConfigNode
+	exportMap   map[string]*ExportConfigNode
+}
+
+// NewConfigNodeMap will create an initial ConfigNodeMap. Append must be called
+// to populate NewConfigNodeMap.
+func NewConfigNodeMap() *ConfigNodeMap {
+	return &ConfigNodeMap{
+		logging:     nil,
+		tracing:     nil,
+		argumentMap: map[string]*ArgumentConfigNode{},
+		exportMap:   map[string]*ExportConfigNode{},
```

Comment:
I'm still generally against us defining a second way to track nodes in the graph, even if it's temporary; I would like for us to find a way to reuse dag.Graph as the source of truth for everything related to config nodes.

Reply:
ConfigNodeMap is been removed